args/audio/systems/audiosystem.inl

```diff
@@ -0,0 +1,273 @@
+#pragma once
+
+namespace args::audio
+{
+	inline void AudioSystem::setup()
+	{
+		m_contextLock = async::readonly_rw_spinlock();
+		async::readwrite_guard guard(m_contextLock);
+
+		alDevice = alcOpenDevice(NULL);
+		if (!alDevice)
+		{
+			// Failed to create alcDevice
+			log::error("OpenAl device failed to open");
+#if defined(AUDIO_EXIT_ON_FAIL)
+			raiseEvent<events::exit>();
+#endif
+			return;
+		}
```

Comment:
In setup everything is still single threaded so locking the lock isn't necessary yet.